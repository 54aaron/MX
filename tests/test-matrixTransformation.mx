int main(){
    Matrix n;
    n = [[1,2],[3,4]];
    prints("==============================================");
    prints("Matrix n:");
    print_matrix(n);
    prints("==============================================");
    prints("Matrix Transformation About y = x");
    print_matrix(transformation(n,1));
    prints("==============================================");
    prints("Matrix Transformation About y = -x");
    print_matrix(transformation(n,2));
    prints("==============================================");
    prints("Matrix Transformation About y-axis");
    print_matrix(transformation(n,3));
    prints("==============================================");
    prints("Matrix Transformation About x-axis");
    print_matrix(transformation(n,4));
    prints("==============================================");
}
