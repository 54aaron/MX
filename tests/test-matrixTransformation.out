==============================================
Matrix n:
[ 1, 2 ]
[ 3, 4 ]

==============================================
Matrix Transformation About y = x
[ 3, 4 ]
[ 1, 2 ]

==============================================
Matrix Transformation About y = -x
[ -3, -4 ]
[ -1, -2 ]

==============================================
Matrix Transformation About x-axis
[ 1, 2 ]
[ -3, -4 ]

==============================================
Matrix Transformation About y-axis
[ -1, -2 ]
[ 3, 4 ]

==============================================
